device/CMakeLists.txt

```diff
@@ -15,18 +16,50 @@ set(UMD_DEVICE_SRCS
     wormhole/wormhole_implementation.cpp
     pcie/pci_device.cpp
 )
-add_library(umd_device SHARED ${UMD_DEVICE_SRCS})
-target_link_libraries(umd_device 
-    PUBLIC yaml-cpp::yaml-cpp umd_common_directories nng uv compiler_flags
-    PRIVATE hwloc rt Boost::interprocess fmt
+
+add_library(device SHARED ${UMD_DEVICE_SRCS})
+add_library(${PROJECT_NAME}::device ALIAS device)
+add_library(${PROJECT_NAME}_device ALIAS device) # For legacy I guess
+
+target_include_directories(
+    device
+    PUBLIC
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/device>
+)
+
+# flatbuffers is public - exposed to tt_metal by tt_simulation_device_generated.h
+# nng is public - exposed to tt_metal by tt_simulation_host.hpp
+target_link_libraries(
+    device
+    PUBLIC
+        nng
+        flatbuffers
```

Comment:
```suggestion
        FlatBuffers::FlatBuffers
```

Reply:
https://github.com/google/flatbuffers/blob/807adb73b20f0afd1cfe92c576339ac8d21592a7/CMakeLists.txt#L716-L722

Google did not do a good job ^

FlatBuffers::FlatBuffers is not a real target, its a couple include directories, and causes problems when we need to export our `libdevice.so` as it is not a part of any export set.

The only way I could get this working was to directly link the target flatbuffers.

I also had to ensure we build it.

We were trying to use flatbuffers as a header only library for some reason, which doesn't look to be officially supported by flatbuffer CMake infra.

Reply:
They have a FlatBuffers target AND a flatbuffers target?  srsy?  🙈